scripts/05_salesrule.sql

```diff
@@ -97,13 +142,26 @@ ALTER TABLE `salesrule_product_attribute`
 -- ----------------
 -- Drop sequence --
 -- ----------------
+-- We need to clean up the salesrule_coupon table before dropping the sequence_salesrule table
+DELETE FROM salesrule_coupon
+    WHERE rule_id NOT IN (SELECT rule_id FROM salesrule);
 
 ALTER TABLE `salesrule_coupon`
     DROP FOREIGN KEY `SALESRULE_COUPON_RULE_ID_SEQUENCE_SALESRULE_SEQUENCE_VALUE`,
     ADD CONSTRAINT `SALESRULE_COUPON_RULE_ID_SALESRULE_RULE_ID` FOREIGN KEY (`rule_id`) REFERENCES `salesrule` (`rule_id`);
 ALTER TABLE `salesrule_customer`
     DROP FOREIGN KEY `SALESRULE_CUSTOMER_RULE_ID_SEQUENCE_SALESRULE_SEQUENCE_VALUE`,
     ADD CONSTRAINT `SALESRULE_CUSTOMER_RULE_ID_SALESRULE_RULE_ID` FOREIGN KEY (`rule_id`) REFERENCES `salesrule` (`rule_id`);
+
+--
+-- Already done above in line 103.
```

Comment:
isn't it line 111?

We should regroup the operation over the same table in the same group of queries for more clarity